pkg/eval/builtin_fn_misc.go

```diff
@@ -309,12 +333,39 @@ func readFileUTF8(fname string) (string, error) {
 	return string(bytes), nil
 }
 
-func sleep(fm *Frame, t float64) error {
-	d := time.Duration(float64(time.Second) * t)
+// This allows for unit tests to efficiently test the behavior of the `sleep`
+// command; both by eliminating an actual sleep and verifying the duration was
+// properly parsed.
+//var TimeAfter func(d time.Duration) <-chan time.Time = func (d time.Duration, fm *Frame) {time.After
+var TimeAfter func(fm *Frame, d time.Duration) <-chan time.Time = func(
+	fm *Frame, d time.Duration) <-chan time.Time {
+	return time.After(d)
+}
+
+func sleep(fm *Frame, duration string) error {
+	r, _ := utf8.DecodeLastRuneInString(duration)
+	if !unicode.IsLetter(r) {
+		// The duration ends with something other than a letter which means it
+		// does not have a time unit suffix. We want seconds to be the default
+		// time unit so append "s" if the string is a simple decimal integer.
+		// We don't want to convert something like "1m3" to "1m3s" since that
+		// is arguably wrong.
+		for _, r := range duration {
+			if !unicode.IsDigit(r) && r != '.' {
+				return fmt.Errorf("invalid sleep duration")
+			}
+		}
+		duration += "s"
+	}
```

Comment:
What about this:

* Try to parse `duration` as a number using `strconv.ParseFloat` first, and if successful, use `time.Duration(float64(time.Second) * t)` (`t` being the parsed float) as the duration

* Otherwise, parse it as a duration using `time.ParseDuration`.

Reply:
SGTM. I originally went with this approach because I hoped that people would start using explicit durations rather than simple number with implied units of seconds. My original approach makes it clear we normally expect a time unit suffix. But even if people start writing `sleep 3s` rather than `sleep 3` I agree it isn't worth the somewhat convoluted logic. And, sadly, I expect that most people will continue to use simple numbers without a time unit suffix.